File: 0301_0400/0310/0310.py
with open("input.txt", "r") as f, open("output.txt", "w") as q:
    for s in f.readlines()[1:]:
        x, y, a = (int(x) for x in s.split())
        q.write('1' if a < 3 or x % a + y % a == 2 else '0')

File: 0301_0400/0310/0310.c
#include <stdio.h>

int main()
{
	FILE *f = fopen("input.txt", "r"), *q = fopen("output.txt", "w");
	int k, x, y, a;

	fscanf(f, "%d", &k);
	while (k-- && fscanf(f, "%d%d%d", &x, &y, &a))
		putc(a < 3 || x % a + y % a == 2 ? '1' : '0', q);
	return 0;
}
